### PR TITLE
Update feature: Improvement priceFormat

### PR DESCRIPTION
**Pull request by GuillaumeSmaha**
Dev: Remove String.fromCharCode: Key map != Ascii map !
Dev: Allow Home & End key
Dev: Add option to allow CtrlKey: default: true
Dev: Add option to keep cursor at the same place except for the decimal part: default: true
Dev: Add option to shift cents into integer: default: false
Dev: Add option to start directly with the integer part: default: true

*It doesn't work on input with int only or thousand separator and int only because priceFormat is called twice in https://github.com/LimeSurvey/LimeSurvey/blob/master/scripts/numerical_input.js#L17

### CODE REVIEW

**Comment by Shnoulle** [comment]
Update the jquery-price-format, then send it to Sam.

For twice call a .not('.integeronly') do the trick :)

**Comment by GuillaumeSmaha** [comment]
@Shnoulle Twice priceFormat should be fixed in the PR https://github.com/LimeSurvey/LimeSurvey/pull/412

**Comment by GuillaumeSmaha** [comment]
This also fix a bug with Previous button.

For example : 2 groups. On the first : 1 numerical input with thousand separator and on the second whatever.
Enter the number 200.00. Click Next
Click Previous: The number is convert into 2.00 due the previous version of priceFormat which always take the 2 last digits as the decimal part


May be https://bugs.limesurvey.org/view.php?id=10211

**Comment by GuillaumeSmaha** [comment]
Bug found : Int only & thousand separator : Too much zeros.

**Comment by GuillaumeSmaha** [comment]
Ok no bug ... Just my test case which doesn't work.
I had twice calls of priceFormat for the inputOnly integer.

**Comment by maziminke** [comment]
Denis, can you look into this and merge if everything looks fine?
Thanks!

**Comment by Shnoulle** [comment]
No ,

Because if i merge , last time there are a bug in it : i'm in cahrge of correction. And actually : i don't want charge on it.

**Comment by maziminke** [comment]
Carsten, do you have balls :-)

Am 19.02.2016 um 10:12 schrieb Denis Chenu:

**Comment by Shnoulle** [comment]
Assigned to @olleharstedt because he start to work on it. There are 2 other pull request for fix thousand_separator ;)

File: third_party/jquery-price-format/jquery_price_format.js [hunk 1]
```diff
@@ -65,7 +74,30 @@
...
+						dObj.selectionStart = pointPosition; 
+						dObj.selectionEnd = pointPosition;
+					}
+					else
+					{
+						obj.val(nvalue);
+					}
+				}
 				else
 					obj.html(nvalue);
```
Comment (olleharstedt) [inline third_party/jquery-price-format/jquery_price_format.js]: Please always add curly braces around all if- and else-statements.

**Comment by olleharstedt** [comment]
If I enter `12.12` into a number field and then click backspace to remove it, the field will be filled with zeros.

File: third_party/jquery-price-format/jquery_price_format.js [hunk 2]
```diff
@@ -201,13 +261,21 @@
 				// Minus Sign, Plus Sign
 				if (allowNegative && (code == 189 || code == 109 || code == 173)) functional = true;
 				if (insertPlusSign && (code == 187 || code == 107 || code == 61)) functional = true;
+
+				// Home, End
+				if (code == 35) functional = true;
```
Comment (olleharstedt) [inline third_party/jquery-price-format/jquery_price_format.js]: As above, please add curly braces.

**Comment by olleharstedt** [comment]
Another issue: If I enter 12 and then click tab to next numerical question and enter 12, the result will be 120.

**Comment by olleharstedt** [comment]
I merged this by mistake, it should still be open. Sorry.

**Comment by GuillaumeSmaha** [comment]
@olleharstedt  Ok, I am fixing it

**Comment by GuillaumeSmaha** [comment]
@olleharstedt If you revert the merge, I can create a new pull request.

**Comment by olleharstedt** [comment]
It's reverted.

There's a lot of work on this feature right now, and it has a lot of issues... Shnoulle just did this one: https://github.com/LimeSurvey/LimeSurvey/pull/445

I would suggest:

1. One fix should be one commit. Now it's many fixes in one, which makes it harder to follow.
2. Please please make sure to link to bugs in the bug-tracker. It's not clear what is bug fix and what is a new feature implementation. If you split your merge requests between bug fixes and new features we can prioritize the bug fixes.

Thanks for your effort.

**Comment by GuillaumeSmaha** [comment]
Ok thanks
It's already splited in few commits :
- https://github.com/LimeSurvey/LimeSurvey/pull/421 This pull request to enhanced the input 
behavior and have a friendly input to use
Now, there is a new pull request : https://github.com/LimeSurvey/LimeSurvey/pull/446
- https://github.com/LimeSurvey/LimeSurvey/pull/407 This aims to fix the drawing of the result : Draw 9 instead of 9.00000 for multiple numerical input in CSV, PDF and HTML
- https://github.com/LimeSurvey/LimeSurvey/pull/412 This fixes the value saved in the database to remove correctly the thousand marks and to convert the decimal mark into a point.
- https://github.com/LimeSurvey/LimeSurvey/pull/408 This fixes an issue with big number (issue with number_format)

**Pull request merged**